### PR TITLE
Porting to ROS2 (New PR)

### PR DESCRIPTION
**Pull request by RoboticsYY**
Hi, since AcutronicRobotics cannot continue to support [PR#11](https://github.com/OctoMap/octomap_msgs/pull/11), this PR is a following effort along AcutronicRobotics' work to port this package to ROS2. Commit history of the original PR is reserved. Any new change requests can be addressed in this PR. PR#11 can be closed.

### CODE REVIEW

**Comment by henningkayser** [comment]
@ahornung @wxmerkt could you give this a review?

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,27 +1,44 @@
...
 
-find_package(catkin REQUIRED COMPONENTS message_generation geometry_msgs std_msgs)
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+
+# Disable Wredundant-decls warnings since rosidl generates redundant function declarations 
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wpedantic -Wno-redundant-decls")
```
Comment (wxmerkt) [inline CMakeLists.txt:11]: Could we use `target_compile_options` or is this not possible due to the use of the `rosidl_generate_interfaces` macro?

**Review by wxmerkt — COMMENTED** [review]
Could we also please backport the whitespace changes and the changes to the message files to minimise diversion of the two branches, and then also drop the unnecessary commits from history (README changes) before this is merged?

File: CMakeLists.txt (hunk 1 above)
Comment (RoboticsYY) [reply CMakeLists.txt:11]: The redundant declaration warnings are brought by the msgs files generated by the `rosidl_generate_interfaces`, thus they cannot be impacted by `target_compile_options` which only adds compile options to the target created by `add_executable()` or `add_library()`.

**Comment by RoboticsYY** [comment]
> Could we also please backport the whitespace changes and the changes to the message files to minimise diversion of the two branches, and then also drop the unnecessary commits from history (README changes) before this is merged?

@wxmerkt The whitespace changes have been reverted. The current changes for the message files are necessary for a successful compilation.

@henningkayser I tried to squash the commits, but seems I cannot push to this branch.

**Comment by henningkayser** [comment]
@wxmerkt I cleaned up the commit history.
@RoboticsYY thanks for preparing this!

**Comment by wxmerkt** [comment]
Thank you @henningkayser :-)

**Pull request merged**